azure-pipelines.yml

```diff
@@ -50,6 +50,9 @@ jobs:
       wheel
     displayName: 'Install deps'
     failOnStderr: true
+  - script: |
+      sudo apt-get install -y wamerican-small
+    displayName: 'Install word list (for csgraph tutorial)'
```

Comment:
Is this package easily available on all platforms and is there a good reason we can't add shims to refguide_check.py to indicate it is not present and either raise an error or skip checking that tutorial?

Reply:
I can override `DocTestRunner.report_unexpected_exception` to handle `FileNotFoundError` differently. But I'm not sure what you'd want to do instead. The error is 

If your problem is just that the error isn't the last in the list, I've found a small shim to stop running the doctest on the first exception. I can even examine the exception to only do this for specific exceptions.

Reply:
I think I'm suggesting the doctests shouldn't run at all (don't even "collect" them) if a required dependency is missing, but maybe not worth the energy to pursue at this point.

Alternatively, a selective exclusion of that tutorial when the dep is missing. A bit like the argument for pytest xfail/skip in selective scenarios--sure, the error could be obvious but what is it really contributing to the test suite vs. a controlled handling when you know what is wrong.